pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -2462,8 +2463,13 @@ spec:
 		Kind:     "Task",
 		TaskSpec: &v1beta1.TaskSpec{Steps: simpleTask.Spec.Steps, Workspaces: simpleTask.Spec.Workspaces},
 	}
-	taskSpec := updateTaskSpecParamsContextsResults(context.Background(), taskRun, rtr)
-	pod, err := r.createPod(testAssets.Ctx, taskSpec, taskRun, rtr)
+	ctx := getContextBasedOnFeatureFlag("alpha")
+	workspaceVolumes := workspace.CreateVolumes(taskRun.Spec.Workspaces)
```

Comment:
Hmm, the fact the we have to have so much logic here in this test isn't great and indicates we might not be testing this at the right level. Wondering if there are any alternatives here e.g. could we call r.reconcile and check that the created pod has the right volumeMounts? Alternatively, we can group this logic into its own function and test that? What do you think?

Reply:
Sure.

Reply:
I have a similar comment to dibyo here- what were the results of your experimentation?

Reply:
I ended up implementing a dedicated unit test `TestUpdateTaskSpecParamsContextsResults` to check the logic until pod creation.

Reply:
I think this test case still has a decent amount of logic in it (getting workspaces from the taskrun spec, calling workspace.CreateVolumes, and passing that into the function being tested), which doesn't really tell us whether the actual code is using this function correctly. I'd prefer to see a test that calls `Reconcile` with a TaskRun spec, and verifies that the output TaskRun has the correct volumes.

Reply:
Sure. I can try to implement that.

Reply:
done.